### PR TITLE
Board View funtional tests

### PR DESCRIPTION
**Pull request by naina-verma**
WI board view tests

### CODE REVIEW

**Comment by naina-verma** [comment]
uilding source...done.
Running tests...
[15:26:46] I/launcher - Running 1 instances of WebDriver
[15:26:46] I/hosted - Using the selenium server at http://localhost:4444/wd/hub
Started
...............................................................................................


95 specs, 0 failures
Finished in 523.687 seconds

[15:35:32] I/launcher - 0 instance(s) of WebDriver still running
[15:35:32] I/launcher - chrome #01 passed
4444/tcp:             1041
8088/tcp:             1071
Functional tests OK

**Comment by naina-verma** [comment]
[test]

File: src/tests/work-item/work-item-list/board.spec.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  constants = require('./constants'),
+  testSupport = require('./testSupport');
+  WorkItemDetailPage = require('./page-objects/work-item-detail.page'),
+  WorkBoardPage = require('./page-objects/work-item-board.page'),
+describe('Work board tests :: ', function () {
+  var page, items, browserMode;
+  page = new WorkItemListPage(true);
+  detailPage = new WorkItemDetailPage(true);
+  var until = protractor.ExpectedConditions;
+  var waitTime = 30000;
```
Comment (ldimaggi) [inline src/tests/work-item/work-item-list/board.spec.js]: Wait time is actually defined in the constants module.

File: src/tests/work-item/work-item-list/board.spec.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    testSupport.setBrowserMode('desktop');
+    boardPage = new WorkBoardPage(true);
+  });
+
+  it('Verify Board elements are present -desktop ', function() {
+        var stateList = typesOfStatesList();
+        var countstateList = totalCountOftypesOfStatesList();
+        verifytypesOfStates(stateList,'false','false');
+        verifytotalCountPertypesOfStates(countstateList,'false','false');
+  });
```
Comment (ldimaggi) [inline src/tests/work-item/work-item-list/board.spec.js]: Can we add a blank line here? It will make the code easier to read. Thx!

**Comment by naina-verma** [comment]
Hi @ldimaggi ,

you are correct we can do loops in the same tests, but due to time out error i removed it and did single tests for selected columns!
sometimes , the tests got stuck in between due to time outs!

File: src/tests/work-item/work-item-list/board.spec.js (hunk 1 above)
Comment (naina-verma) [inline src/tests/work-item/work-item-list/board.spec.js]: Done

File: src/tests/work-item/work-item-list/board.spec.js (hunk 2 above)
Comment (naina-verma) [inline src/tests/work-item/work-item-list/board.spec.js]: Done

File: src/tests/work-item/work-item-list/board.spec.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        browser.wait(until.elementToBeClickable(page.spaceDropdown), constants.WAIT, 'Failed to find elemet'); 
+      
+        page.clickOnSpaceDropdown();
+        page.selectSpaceDropDownValue("3");
+       
+        browser.wait(until.elementToBeClickable(page.openButton), constants.WAIT, 'Failed to find elemet'); 
+        var countstateList = totalCountOftypesOfStatesList();
+        verifytotalCountPertypesOfStates(countstateList,'1','1'); 
+
+  });
```
Comment (naina-verma) [inline src/tests/work-item/work-item-list/board.spec.js]: Done

**Pull request merged**